diff -ur tenderlove-nokogiri-17c2ced-o/lib/nokogiri.rb tenderlove-nokogiri-17c2ced/lib/nokogiri.rb
--- tenderlove-nokogiri-17c2ced-o/lib/nokogiri.rb	2010-08-28 11:56:17.000000000 -0600
+++ tenderlove-nokogiri-17c2ced/lib/nokogiri.rb	2010-08-28 13:05:45.000000000 -0600
@@ -10,7 +10,7 @@
   require 'ffi'
   require 'nokogiri/ffi/libxml'
 else
-  require 'nokogiri/nokogiri'
+  require 'nokogiri.so'
 end
 
 warn(<<-eowarn) if '1.8.6' == RUBY_VERSION && $VERBOSE
